src/assets/js/customizer/buttons/preview.js

```diff
@@ -0,0 +1,236 @@
+const api      = wp.customize;
+const colorLib = window.net.brehaut;
+
+/**
+ * File: src/assets/js/customizer/buttons/preview.js
+ *
+ * Preview utility for the buttons.
+ *
+ * @since 2.12.0
+ * @package BGTFW/customizer
+ * @author  BoldGrid
+ */
+
+/**
+ * Class: Preview
+ *
+ * Preview utility for the buttons.
+ *
+ * @since 2.12.0
+ */
+export class Preview {
+	constructor() {
+
+		/**
+		 * Array of button Types used to form control ids.
+		 * Ex: bgtfw_primary_button_background.
+		 */
+		this.buttonTypes = [ 'primary', 'secondary' ];
+
+		/**
+		 * Array of button controls that all use the same method
+		 * of generating preview classes in the following manner:
+		 *
+		 * Key:   control suffix.
+		 * Value: array of classes to remove.
+		 */
+		this.generalButtonControls = {
+			'raised': [ 'btn-raised' ],
+			'text_shadow': [ 'btn-longshadow' ],
+			'effect': [ 'btn-3d', 'btn-glow' ],
+			'border': [ 'btn-border', 'btn-border-thin', 'btn-border-thick' ],
+			'shape': [ 'btn-rounded', 'btn-pill' ]
+		};
+	}
+
+	/**
+	 * Get Hover Color.
+	 *
+	 * This class auto generates a ligher or darker
+	 * hover color for the button's background color.
+	 *
+	 * @since 2.12.0
+	 *
+	 * @param   {string} color color string.
+	 * @returns {string}       css formatted color string.
```

Comment:
Not sure on this one, but the description shouldn't have the extra whitespace after string.